File: ruby/exec/update_wiki_list_on_home_and_sidebar.rb
require_relative '../src/home'
require_relative '../src/sidebar'

group_by, language, home_overflow, *_ = ARGV
params = { group_by:, language:, home_overflow: }.reject { |_, value| value.empty? }

puts '-------------------- Categorising the Entire github-wiki-organisers Wiki Pages... --------------------'
puts "\n"
puts '-------------------- Organising Home... --------------------'
home_url = Home.run(**params)
puts "Check out an Up-to-date Wiki List on Home at '#{home_url}' !!"
puts '-------------------- Done Organising Home 🎉 --------------------'
puts "\n"
puts '-------------------- Organising Sidebar... --------------------'
Sidebar.run(**params)
puts "Check out an Up-to-date Wiki List on Sidebar at '#{home_url}' !!"
puts '-------------------- Done Organising Sidebar 🎉 --------------------'
puts "\n"
puts '-------------------- Done Categorising the Entire github-wiki-organisers Wiki Pages 🎉 --------------------'

File: python/exec/update_wiki_list_on_home_and_sidebar.py
from sidebar import Sidebar
from home import Home
import sys
import os
import shutil
import glob
sys.path.append('./src')


_, group_by, language, home_overflow, *_ = sys.argv
params = dict()
for key, value in {'group_by': group_by, 'language': language,
                   'home_overflow': home_overflow}.items():
    if value:
        params[key] = value

print('-------------------- Categorising the Entire github-wiki-organisers Wiki Pages... --------------------')
print()
print('-------------------- Organising Home... --------------------')
home_url = Home(**params).run()
print(f"Check out an Up-to-date Wiki List on Home at '{home_url}' !!")
print('-------------------- Done Organising Home 🎉 --------------------')
print()
print('-------------------- Organising Sidebar... --------------------')
Sidebar(**params).run()
print(f"Check out an Up-to-date Wiki List on Sidebar at '{home_url}' !!")
print('-------------------- Done Organising Sidebar 🎉 --------------------')
print()
print('-------------------- Done Categorising the Entire github-wiki-organisers Wiki Pages 🎉 --------------------')

for pycache in glob.glob(
        os.path.join(
            '.',
            '**',
            '__pycache__'),
        recursive=True):
    if os.path.exists(pycache):
        shutil.rmtree(pycache)
